Model/AjaxResponse.php

```diff
@@ -208,9 +206,9 @@ protected function getFilterItems(): array
                 $items[$datascope] = [];
                 $attribute = $filter->getAttributeModel();
                 foreach ($filter->getItems() as $item) {
-                    if($this->swatchHelper->isSwatchAttribute($attribute)) {
+                    if ($this->swatchHelper->isSwatchAttribute($attribute)) {
```

Comment:
The call to `isSwatchAttribute` will be made multiple times, because it's with the `foreach ($filter->getItems() as $items)` loop.
I would suggest to pull up this `if` statement, so you check if it's a swatch attribute only once per each filter, and not multiple times per each filter (meaning, one time per each filter item).

Maybe something like that:

```php

              $attribute = $filter->getAttributeModel();
              if ($this->swatchHelper->isSwatchAttribute($attribute)) {
                  foreach ($filter->getItems() as $item) {
                      // prepare the swatch'ed item
                  }
              } else {
                  foreach ($filter->getItems() as $item) {
                        $items[$datascope][] = $item->toArray(['label', 'count', 'url', 'is_selected']);
                  }
              }

```